x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/hooks/use_monitor_latest_ping.tsx

```diff
@@ -28,10 +29,14 @@ export const useMonitorLatestPing = (params?: UseMonitorLatestPingParams) => {
   const latestPing = useSelector(selectLatestPing);
   const pingsLoading = useSelector(selectPingsLoading);
 
-  const isUpToDate =
-    latestPing &&
-    latestPing.monitor.id === monitorId &&
-    latestPing.observer?.geo?.name === locationLabel;
+  const latestPingId = latestPing?.monitor.id;
+
+  const isIdSame =
```

Comment:
Take into account that both `latestPingId` and `monitorId` might be `undefined`, and in that case this will evaluate to `true`.